webapp/src/pages/Users/Users.tsx

```diff
@@ -1,73 +1,99 @@
+import './Users.css';
 import {
-  Header,
   IconVariant,
+  Input,
+  InputVariant,
   NavigationBar,
   RowsList,
+  RowsListProps,
+  SmallAvatar,
 } from '../../shared/components';
-import './Users.css';
+import React, { useEffect, useState } from 'react';
+import {
+  USER_URL,
+  USERS_EP_URL,
+  USERS_URL,
+  WILDCARD_AVATAR_URL,
+} from '../../shared/urls';
+import { Link } from 'react-router-dom';
+
+class User {
+  username: string;
+  email: string;
+  avatarId: string | null;
+  id: string;
+  createdAt: Date;
+
+  constructor(
+    username: string,
+    email: string,
+    avatarId: string | null,
+    id: string,
+    createdAt: Date,
+  ) {
+    this.username = username;
+    this.email = email;
+    this.avatarId = avatarId;
+    this.id = id;
+    this.createdAt = createdAt;
+  }
+}
 
 export default function Users() {
-  const buttonAction = (): void => alert('This is an alert');
-  const buttonLink = (): void => {
-    window.location.href = 'https://google.com';
+  const [usersList, setUsersList] = useState<RowsListProps>({ rows: [] });
+
+  const getRows = async (url: string): Promise<RowsListProps> => {
+    let rows: RowsListProps = { rows: [] };
+    const response = await fetch(url);
+    const users: User[] = (await response.json()) ?? [];
+    users.forEach((user) => {
+      rows.rows?.push({
+        iconVariant: IconVariant.ARROW_FORWARD,
+        avatarProps: {
+          url: user.avatarId
+            ? `${USERS_EP_URL}/${user.id}/avatar`
+            : WILDCARD_AVATAR_URL,
+          status: 'offline',
+        },
+        onClick: () => {
+          window.location.href = `${USERS_URL}/${user.username}`;
+        },
+        title: user.username,
+        subtitle: 'level x',
+        key: user.id,
+      });
+    });
+    return rows;
   };
-  const randomAvatar = 'https://i.pravatar.cc/1000';
-  const randomAvatar2 = 'https://i.pravatar.cc/2000';
-  const randomAvatar3 = 'https://i.pravatar.cc/3000';
+
+  useEffect(() => {
+    const fetchUsersList = async () => {
```

Comment:
Good job for connecting it to the backend :rocket:

Reply:
thank you